### ISSUE TITLE
Syntax for toThrow?

### ISSUE THREAD

**Issue by provegard** [factor10/intent#28]
For example:

```
expect(<...>).toThrow[IllegalArgumentException]()
```

With message (regex?)

```
expect(<...>).toThrow[IllegalArgumentException]("it failed")
```

Or with pattern matching:

```
expect(<...>).toThrow:
  case ex: IllegalArgumentException => expect(ex.getMessage).toMatch(...)
```

But there are two expects there... We could do it like `whenComplete`:

```
expectFailure(<...>):
  ex => ... // could do pattern matching here also
```

Or something like `intercept` in ScalaTest.

@eliasson thoughts?

**Comment by eliasson** [comment]
> For example:
> With message (regex?)
> 
> ```
> expect(<...>).toThrow[IllegalArgumentException]("it failed")
> ```
> 

This one. I often find that you want to make some assert on type _and_ message (type alone is too weak IMO)

And if you need pattern matching in an assert you really should split the test into multiple tests.

**Comment by provegard** [comment]
Fixed in #37
